org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/engine/connected/ConnectedEngineFacade.java

```diff
@@ -214,7 +213,7 @@ private void doWithEngine(Consumer<ConnectedSonarLintEngine> consumer) {
 
   private void reloadProjects(ConnectedSonarLintEngine engine) {
     this.allProjectsByKey.clear();
-    this.allProjectsByKey.putAll(engine.allProjectsByKey());
+    this.allProjectsByKey.putAll(engine.downloadAllProjects(createEndpointParams(), buildClientWithProxyAndCredentials(), null));
```

Comment:
Could this block the UI thread ? Also the `reloadProjects` method is called elsewhere in the class just after downloading all projects, so we could do it twice in a row